app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+  
+  def new
+    redirect_to home_path if logged_in?    
+  end
+
+  def create
+    user = User.find_by(:email_address => params[:email_address])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      flash[:notice] = "You have successfully logged in!"
+      redirect_to home_path
+    else
+      flash[:error] = "There was a problem with your log in"
```

Comment:
It should be `flash.now` when you render. Otherwise it sticks around for another request.